### ISSUE TITLE
False-positive diagnostic when comparing union with primitive types

### ISSUE THREAD

**Issue by georgiy-belyanin** [EmmyLuaLs/emmylua-analyzer-rust#57]
There are a lot of false-positives when comparing union types with others. For instance see the following.

```lua
---@type 1 | 2 | 3
local a = 2

---@param a number
local function test(a)
end

--False-positive warning, number is not compatible with 1 | 2 | 3
test(a)
```
